### ISSUE TITLE
Too much Distortion while synthesizing high pitch female voices

### ISSUE THREAD

**Issue by bharaniyv** [DigitalPhonetics/IMS-Toucan#53]
Hi,

I really liked the prosody transfer capabilities of the new GST based model, but it is not able to handle all types of voices like the ECAPA model, for example I tried synthesizing some High pitch female voices and Female mocking emotion samples. The output is too distorted, I even tried fine tuning the Style Embedding layer using samples with Background noise and High pitch voices (used sox to increase pitch of a female tts dataset), there is a slight improvement but the problem still exists.

I noticed that you are working an a bigger embedding func with some modification in a separate branch, I just wanted to know if you have noticed this issue and if you did how is the new model performing in these cases and is there any projected date for the release of the new model?

Thanks

**Comment by Flux9665** [comment]
Yes, I noticed that female voices can sound pretty strange if the reference has an emotion with lots of arousal, like anger. With the bigger embeddings and the new ways for integrating the embeddings it's a lot better. But it still does not sound clean enough. Some data augmentation, as you suggest, might be interesting. Also, I have some more ideas on how to extract the embeddings. The models are pretty much done training, just the GAN for making speaker embeddings controllable has some problems with the bigger embeddings and is still under construction. I hope by the end of the week I can update the most recent release with the better models.

**Comment by bharaniyv** [comment]
Thanks for the update, I too trained a speaker embedding function based on your new code and as you mentioned the female voice is a bit better but not perfect even with the data augmentation the quality is still lacking for female voices especially for unseen female high pitch / emotional voices, but I am hoping your model might perform better since I trained my model on a small batch size of 16 and I think a bigger batch size might help little bit more.

**Comment by bharaniyv** [comment]
I am just wondering have you considered combining the old prosody model of using Speaker recognition embeddings (preferably TDNN only without ECAPA since ECAPA is too dominating in my old custom trained models) and new GST embeddings and using them for the training, I am thinking it might help in adapting to unknown and high emotional reference sample better, since TDNN can provide stability and GST can provide more matured prosody. Let me know what are your thoughts about this approach, do you think it is worth trying?

**Comment by Flux9665** [comment]
It could be interesting to remove the speaker information from GST to get prosody-only embeddings. The domain adversarial learning / gradient reversal process could probably achieve this. I don't have the time to do it right now because I have to do a lot of teaching, but in the future I might try that. If your try something like this, please let me know! I'm just worried that without methods like domain adversarial learning, the two embeddings would learn complementary stuff. I also considered pretraining GST with speaker identification, but that would require labels and might induce biases, so I'm not sure whether it's a good idea. Leaving the model to figure out on its own what it needs makes the most sense to me intuitively.

**Comment by bharaniyv** [comment]
Thanks for the clarification, I will let you know if you I try GST + X-Vector model and have you tried integrating fundamental frequency (f0) in the GST I think it can also help to solve unseen emotions issue, meanwhile I am eager to check the quality of the new Expanded embedding model performance on the female emotional samples, when can we expect a model release and is the code stable enough to train on my own?

**Comment by Flux9665** [comment]
The code should be pretty stable, I haven't found a bug for a while now. But I'm probably going to try out something else, because we ran into some problems with the embedding-generation GAN and the bigger embeddings. It's hard to find a low-rank projection for them apparently. 

I put the models that work with the current commit of the experimental branch (https://github.com/DigitalPhonetics/IMS-Toucan/tree/2faeea316e9248eef3855d7fbda0c0e11558215b) here https://drive.google.com/drive/folders/1b4hcVTsUt2Z1Rh2MbHcxZuP5xXupVRwd?usp=sharing

Official release will wait until we found a setup that works better with the GAN.

**Comment by bharaniyv** [comment]
Thanks for the models I will try out the new models and also I started working on GST + x-vector model based on the ESPNET FastSpeech2 implementation. I am more interested in cloning prosody with better quality than GAN model so I want to try if GST + X-vector model can work and give better prosody quality

**Comment by Flux9665** [comment]
new models with new speaker embeddings are in the current release, should hopefully be fixed.
